server/src/main/java/org/elasticsearch/index/mapper/MapperMetrics.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+/**
+ * Groups together all metrics used in mappers.
+ * Main purpose of this class is to avoid verbosity of passing individual metric instances around.
+ */
+public record MapperMetrics(SourceFieldMetrics sourceFieldMetrics) {
```

Comment:
I wonder if we should make this even more generic given that it is now in `SearchExecutionContext`.

Reply:
I think this is fine, but I'd use a container like `SearchExecutionMetrics` to the context and include `MapperMetrics` in it.